staging/src/k8s.io/component-base/cli/flag/sectioned.go

```diff
@@ -40,12 +42,22 @@ func (nfs *NamedFlagSets) FlagSet(name string) *pflag.FlagSet {
 		nfs.FlagSets = map[string]*pflag.FlagSet{}
 	}
 	if _, ok := nfs.FlagSets[name]; !ok {
-		nfs.FlagSets[name] = pflag.NewFlagSet(name, pflag.ExitOnError)
+		flagSet := pflag.NewFlagSet(name, pflag.ExitOnError)
+		flagSet.SetNormalizeFunc(pflag.CommandLine.GetNormalizeFunc())
+		if nfs.normalizeNameFunc != nil {
+			flagSet.SetNormalizeFunc(nfs.normalizeNameFunc)
+		}
+		nfs.FlagSets[name] = flagSet
 		nfs.Order = append(nfs.Order, name)
 	}
 	return nfs.FlagSets[name]
 }
 
+// SetNormalizeFunc allows you to add a function which used to initialize FlagSets created by NamedFlagSets.
+func (nfs *NamedFlagSets) SetNormalizeFunc(n func(f *pflag.FlagSet, name string) pflag.NormalizedName) {
```

Comment:
the other data fields are already public, simpler to have normalizeNameFunc public rather than a setter